core/templates/dev/head/services/contextual/WindowDimensionsService.ts

```diff
@@ -16,26 +16,37 @@
  * @fileoverview Service for computing the window dimensions.
  */
 
-angular.module('oppia').factory('WindowDimensionsService', [
-  '$window', function($window) {
-    var onResizeHooks = [];
-    angular.element($window).bind('resize', function() {
-      onResizeHooks.forEach(function(hookFn) {
-        hookFn();
-      });
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HostListener, Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WindowDimensionsService {
+  onResizeHooks: Array<any> = [];
+
+  @HostListener('window:resize', ['$event'])
+  onResize(event: Event): void {
+    this.onResizeHooks.forEach(function(hookFn) {
+      hookFn();
     });
-    return {
-      getWidth: function() {
-        return (
-          $window.innerWidth || document.documentElement.clientWidth ||
-          document.body.clientWidth);
-      },
-      registerOnResizeHook: function(hookFn) {
-        onResizeHooks.push(hookFn);
-      },
-      isWindowNarrow: function() {
-        var NORMAL_NAVBAR_CUTOFF_WIDTH_PX = 768;
-        return this.getWidth() <= NORMAL_NAVBAR_CUTOFF_WIDTH_PX;
-      }
-    };
-  }]);
+  }
+
+  getWidth(): number {
+    return ( window.innerWidth || document.documentElement.clientWidth ||
+        document.body.clientWidth);
+  }
+
+  registerOnResizeHook(hookFn: any): void {
```

Comment:
Are we unsure what the type is for the arg or can it actually be anything?

Reply:
@kevinlee12 as we can see in line number 28-31 hookFn is being called as a function 
```
this.onResizeHooks.forEach(function(hookFn) {
      hookFn();
});
```

So i am pretty sure it's a **Function** but I've used **any** just to be safe. 
@bansalnitish should i change it's type to **Function**

Reply:
@bansalnitish let's go ahead and do that.

Reply:
sorry, I meant to tag @Showtim3 :)

Reply:
Done!